### PR TITLE
feat: add call-hierarchy.d.ts

### PR DESCRIPTION
**Pull request by ayame113**
This is needed for https://github.com/atom-community/atom-ide-outline/pull/131.

### CODE REVIEW

File: types-packages/call-hierarchy.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  data: CallHierarchyItem[]
+  /** Returns CallHierarchy data about the i-th CallHierarchyItem. */
+  itemAt(i: number): Promise<CallHierarchy<T>>
+}
+
+interface CallHierarchyItem {
+  path: IdeUri
+  name: string
+  /** Same as the icon of outline.d.ts */
+  icon: string | null
```
Comment (aminya) [inline types-packages/call-hierarchy.d.ts:25]: I prefer undefined for optional properties.
```suggestion
  icon?: string
```

**Review by aminya — CHANGES_REQUESTED** [review]
Could you reexport it from [main.d.ts](https://github.com/atom-community/atom-ide-base/blob/master/types-packages/main.d.ts) too?

**Comment by ayame113** [comment]
Thank you for your review. Fixed.😊

**Review by aminya — APPROVED** [review]
Thank you!

**Comment by aminya** [comment]
Released it in atom-ide-base 3.3.0.

**Pull request merged**